### ISSUE TITLE
VS2019-integrated OpenAPI client support fails for multitargeted assemblies

### ISSUE THREAD

**Issue by lscorcia** [dotnet/aspnetcore#31996]
Minimal repro project here: https://github.com/lscorcia/aspnetcore-issue31996-repro

I have a library which is using multitargeting - in the .csproj it has:
```
    <TargetFrameworks>netstandard2.1;net472</TargetFrameworks>
```
I added to it an OpenAPI connected service via VS2019 GUI (16.9.4):

![immagine](https://user-images.githubusercontent.com/16761934/115464468-695f1880-a22d-11eb-8f36-ad9ef2e7ec9e.png)

It added some nuget packages to the project, and the OpenApiReference element:

```
  <ItemGroup>
    <OpenApiReference Include="OpenAPIs\dss.json">
      <CodeGenerator>NSwagCSharp</CodeGenerator>
    </OpenApiReference>
  </ItemGroup>
...
 <PackageReference Include="Microsoft.Extensions.ApiDescription.Client" Version="5.0.5">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
    <PackageReference Include="Newtonsoft.Json" Version="12.0.2" />
    <PackageReference Include="NSwag.ApiDescription.Client" Version="13.10.9">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
```

The build completes fine, no error or warnings, however it does not generate the corresponding source code in the `obj` folder.
If I replace the multitargeting with a single target however it works correctly:

```
    <TargetFramework>netstandard2.1</TargetFramework>
```

Looking at the build logs, it seems like the following condition fails and does not run the OpenAPI code generation:

https://github.com/dotnet/aspnetcore/blob/ec69111cd568dbc7b06f62851bea3b452e404972/src/Tools/Extensions.ApiDescription.Client/src/build/Microsoft.Extensions.ApiDescription.Client.targets#L161-L164

Relevant [msbuild.log](https://github.com/dotnet/aspnetcore/files/6346971/msbuild.log) entry (in Italian, but should be easily readable):

```
Destinazione"_TieInGenerateOpenApiCode" ignorata. Condizione false. ( '$(OpenApiGenerateCodeOnBuild)' == 'true' AND ('$(TargetFramework)' == '' OR '$(TargetFrameworks)' == '') ) ha restituito ( 'true' == 'true' AND ('netstandard2.1' == '' OR 'netstandard2.1;net472' == '') ).
...
Destinazione"_TieInGenerateOpenApiCode" ignorata. Condizione false. ( '$(OpenApiGenerateCodeOnBuild)' == 'true' AND ('$(TargetFramework)' == '' OR '$(TargetFrameworks)' == '') ) ha restituito ( 'true' == 'true' AND ('net472' == '' OR 'netstandard2.1;net472' == '') ).
```

I can see why one wouldn't want to run the generation multiple times, but it does not run at all.

**Comment by captainsafia** [comment]
Yep, this is a bug in the MSBuild condition here. It's valid to avoid multiple invocations for multiple targets but this disables it all together. I think a condition like:

```
Condition="'$(TargetFrameworks)' == '' OR $(TargetFrameworks.EndsWith($(TargetFramework)))" 
```

Get us a little closer to doing the invocation for only the TFM That appears first. We might have to tweak it a bit to account for projects that target more than 2 TFMs.

**Comment by Sinhk** [comment]
Not sure if it's related to this issue, but it appears to be a typo here:
https://github.com/dotnet/aspnetcore/blob/e7a85fb5de16cb888783537f2374d16bcb9678dc/src/Tools/Extensions.ApiDescription.Client/src/buildMultiTargeting/Microsoft.Extensions.ApiDescription.Client.targets#L18
with two apostrophes before $(OpenApiGenerateCodeOnBuild)

**Comment by AdelRefaat** [comment]
I had to workaround it by setting `TargetFrameworks` to single/current value before running `GenerateOpenApiCode` during compilation
Sample here: https://gist.github.com/AdelRefaat/dc3258c86e2d6e8a06fbff921c0baf96

**Comment by Youssef1313** [comment]
`Microsoft.Extensions.ApiDescription.Client` was deprecated in .NET 10 and dropped in .NET 11.

See https://github.com/dotnet/aspnetcore/issues/68203

**Comment by arknu** [comment]
How is this not fixed? Multitargeting is a fairly common scenario and this is a built-in Visual Studio feature. It is still very much an issue on .NET 10 and VS2026.

**Comment by Youssef1313** [comment]
@arknu Please clarify the issue you are facing. If you are using `Microsoft.Extensions.ApiDescription.Client`, that's already deprecated in .NET 10.

If your issue doesn't involve `Microsoft.Extensions.ApiDescription.Client`, please open a new issue with a complete repro project.

**Comment by arknu** [comment]
It involves using the built-in Visual Studio feature to add a connected service using and OpenAPI file. This will automatically add the NSwag Toolchain and the `NSwag.ApiDescription.Client` package, and the appropriate `OpenApiReference` in the project file. This all works perfectly when using a single target framwork, but the moment you add multitargeting, it breaks and doesn't generate code, exactly as described. Removing the `Microsoft.Extensions.ApiDescription.Client` package makes no difference.

**Comment by Youssef1313** [comment]
Please open a new issue with detailed reproduction steps. The original issue here was specifically involving Microsoft.Extensions.ApiDescription.Client, and hence was closed as not planned as it's deprecated.

**Comment by arknu** [comment]
@Youssef1313 It may be deprecated in .NET 11, but `Microsoft.Extensions.ApiDescription.Client` is still referenced by a feature in the shipping version of Visual Studio, as well as the insider version. I looked a bit further, and `NSwag.ApiDescription.Client` actually has a dependency on `Microsoft.Extensions.ApiDescription.Client`, so it was being included regardless. Hence this issue is still relevant. 

Also, are you saying that using the OpenAPI reference feature in VS will add a deprecated package once .NET 11 is released? Seems like somewhat of an oversight. What is the replacement supposed to be, if NSwag still has a dependency on it.

**Comment by Youssef1313** [comment]
This looks like something to report to Visual Studio as they are using a deprecated package.

**Comment by arknu** [comment]
@Youssef1313 Can we agree that the currently shipping version of this package is _**not**_ deprecated and this is thus an issue that needs to be fixed? I don't care what .NET 11 will do, since that is not releasing until November.

<img width="1119" height="607" alt="Image" src="https://github.com/user-attachments/assets/c1eedf10-c06f-4202-aae2-a780bec4fe7a" />

**Comment by Youssef1313** [comment]
@wtgodbe We need to manually mark 10.0.12 as deprecated on nuget.org, similar to all the previous 10.0.x versions. It looks like we will have to do it manually every month?

**Comment by arknu** [comment]
@Youssef1313 The use of `Microsoft.Extensions.ApiDescription.Client` and NSwag is not confined to Visual Studio, the Microsft.dotnet-openapi command tool described here https://learn.microsoft.com/en-us/aspnet/core/fundamentals/openapi/openapi-tools?view=aspnetcore-10.0 also uses NSwag for code generation using the same infrastructure as Visual Studio, thus including this deprecated package. 

What is the alternative to generating a C# client from an OpenAPI document supposed to be, exactly? Because every officially documented way currently uses a broken package that is apparently deprecated...

**Comment by Youssef1313** [comment]
See https://learn.microsoft.com/en-us/aspnet/core/breaking-changes/10/apidescription-client-deprecated

**Comment by arknu** [comment]
@Youssef1313 So you went from "let's integrate support for generating OpenAPI documents directly into ASP.NET" to "let's also remove any officially supported way to build a client using those OpenAPI documents". How does adding support for generating OpenAPI documents and then _removing_ support for consuming them make any sense? If anything, you should be integrating support for consuming OpenAPI directly, since I fully agree that NSwag is quite bothersome to work with.
